src/compositemark/callout.ts

```diff
@@ -0,0 +1,131 @@
+import {Config} from '../config';
+import {Encoding} from '../encoding';
+import * as log from '../log';
+import {MarkConfig, MarkDef} from '../mark';
+import {GenericUnitSpec, LayerSpec} from '../spec';
+
+
+export const CALLOUT: 'callout' = 'callout';
+export type CALLOUT = typeof CALLOUT;
+
+export interface CalloutDef {
+  type: CALLOUT;
+  calloutAngle?: number;
+  calloutOffset?: number;
+  calloutLength?: number;
+  labelOffset?: number;
+  callout?: MarkDef;
+  label?: MarkDef;
+}
+
+export function isCalloutDef(mark: CALLOUT | CalloutDef): mark is CalloutDef {
+  return !!mark['type'];
+}
+
+export type CalloutStyle = 'callout-label' | 'callout-rule';
+
+export const CALLOUT_STYLES: CalloutStyle[] = ['callout-label', 'callout-rule'];
```

Comment:
We should decide if we want dash or camelCase and be consistent across all composite marks.  

- camelCase is good as this will be a property `config.style.calloutLabel` -- having dash will be awkward?  
- but normally we should dash for property names.  Also we should just redirect `config.callout.label` to `config.style.calloutLabel` / `config.style["callout-label"]` anyway 
   - the question is do we do `config.callout.rule` or just `config.callout` for the part that will get redirected to `config.style.calloutRule` / `config.style["callout-rule"]` 

@domoritz  -- any opinion?